src/commands/amend.rs

```diff
@@ -0,0 +1,70 @@
+//! Amend the current commit.
+//!
+//! This command automatically amends the HEAD commit with changes to files
+//! that are already tracked in the repo.
+//!
+//! Optionally, the user can trigger an automatic restack post-amend with `--restack`,
+//! or by setting the `branchless.amend.restack` config variable.
+
+use std::time::SystemTime;
+
+use eyre::Context;
+
+use crate::commands::gc::mark_commit_reachable;
+use crate::commands::restack;
+use crate::core::config::get_amend_restack;
+use crate::core::eventlog::{Event, EventLogDb};
+use crate::git::{GitRunInfo, Repo};
+use crate::opts::MoveOptions;
+use crate::tui::Effects;
+
+/// Amends the existing HEAD commit.
+pub fn amend(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    restack: bool,
+    move_options: &MoveOptions,
+) -> eyre::Result<isize> {
+    let now = SystemTime::now();
+    let repo = Repo::from_current_dir()?;
+    let conn = repo.get_db_conn()?;
+    let mut event_log_db = EventLogDb::new(&conn)?;
+
+    let head_oid = match repo.get_head_info()?.oid {
+        Some(oid) => oid,
+        None => eyre::bail!("No HEAD to amend to"),
+    };
+    let head = repo.find_commit_or_fail(head_oid)?;
+
+    let mut index = repo.get_index()?;
+    let tree_oid = index.update_all(&["*"])?;
```

Comment:
> you mean executing git as a subprocess, and then parsing the `status` output

Yes, exactly.

> If we do that, and get a list of files that need to be updated in the index, can we call `update_all` on that list, or would that still have performance issues?

We could and it would work in a performant way, but we won't need to, because `git status` updates the index file on disk as a side-effect.